src/java/org/apache/cassandra/service/DataResolver.java

```diff
@@ -860,6 +858,10 @@ public UnfilteredRowIterator moreContents()
                 ColumnFamilyStore.metricsFor(metadata.cfId).shortReadProtectionRequests.mark();
                 Tracing.trace("Requesting {} extra rows from {} for short read protection", lastQueried, source);
 
+                // If we've arrived here, all responses have been consumed, and we're about to request more. Before that
+                // happens, clear the accumulator and allow garbage collection to free the resources they used.
+                responses.clearUnsafe();
```

Comment:
Perhaps it would actually make more sense to just null out the accumulator entries as they're retrieved in `shortReadProtectedResponse()`.

Reply:
...or at least clear only the response from this source, and then probably in `ShortReadPartitionsProtection#moreContents()`. There's another wrinkle here, which is that the short read protection classes are moved out of `DataResolver` in `trunk`. (Maybe we can provide a callback that clears the responses for the appropriate source to the SRP objects...)

CC @adelapena